core/parachain/availability/recovery/recovery_impl.cpp

```diff
@@ -35,7 +35,7 @@ namespace {
     BOOST_ASSERT_MSG(                                                       \
         std::find(results.begin(), results.end(), result) != results.end(), \
         "Unknown result type");                                             \
-    full_recoveries_finished_[strategy][result]->inc();                     \
+    full_recoveries_finished_.at(strategy).at(result)->inc();               \
```

Comment:
This is intended behavior.
`at()` must throw, because `assert`/`BOOST_ASSERT` don't work.

> instantiate at first increment

`operator[]` will insert `nullptr` and cause NullPointerException.